jp2/src/lib.rs

```diff
@@ -1502,8 +1487,45 @@ impl JBox for PaletteBox {
         &mut self,
         reader: &mut R,
     ) -> Result<(), Box<dyn error::Error>> {
-        reader.read_exact(&mut self.num_entries)?;
-        reader.read_exact(&mut self.num_components)?;
+        let mut num_entries_bytes = [0u8; 2];
+        reader.read_exact(&mut num_entries_bytes)?;
+        let num_entries = u16::from_be_bytes(num_entries_bytes);
+
+        let mut num_palette_columns_bytes = [0u8; 1];
+        reader.read_exact(&mut num_palette_columns_bytes)?;
+        let num_palette_columns = u8::from_be_bytes(num_palette_columns_bytes);
+
+        let mut bit_depth_bytes = [0u8; 1];
+        for _ in 0..num_palette_columns {
+            reader.read_exact(&mut bit_depth_bytes)?;
+            self.bit_depths.push(BitDepth::new(bit_depth_bytes[0]));
+        }
+        for _ in 0..num_entries {
+            let mut entry_components = Vec::<u32>::with_capacity(num_palette_columns as usize);
+            for i in 0..num_palette_columns as usize {
+                let num_bytes = self.bit_depths[i].num_bytes() as usize;
+                let value = match num_bytes {
+                    1 => {
+                        let mut value_bytes = [0u8; 1];
+                        reader.read_exact(&mut value_bytes)?;
+                        u8::from_be_bytes(value_bytes) as u32
+                    }
+                    2 => {
+                        let mut value_bytes = [0u8; 2];
+                        reader.read_exact(&mut value_bytes)?;
+                        u16::from_be_bytes(value_bytes) as u32
+                    }
+                    _ => unimplemented!(
+                        "more than 16 bit data is not yet supported for palette entries"
+                    ),
+                };
+                entry_components.push(value);
+            }
+            self.entries.push(entry_components);
+        }
+        /*  TODO
```

Comment:
I think it is time to get rid of this TODO.

Reply:
Yes. My bad.